### ISSUE TITLE
[V1 & 3 Components] Implement available components for Income Limits

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#17276]
## Status
**[2024-06-10]** _**Moved back to Blocked > Randi noted that DST has not yet released this component (though the work has been merged/closed). Once it is available (see releases here: https://github.com/department-of-veterans-affairs/component-library/releases) we can pull this ticket in.**_
**[2024-06-05]** Ticket 2500 has been completed and close. Unblocking this ticket and moving it to Stretch/Next Sprint.
**[2024-05-30]** This ticket is still blocked. The DST ticket is in progress [Link - Action Web Component - Development #2500](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2500) 
**[2024-05-21]** BLOCKED BY department-of-veterans-affairs/vets-design-system-documentation#2500
**[2024-05-15]** BLOCKED BY department-of-veterans-affairs/vets-design-system-documentation#2500
**[2024-05-07]** BLOCKED BY department-of-veterans-affairs/vets-design-system-documentation#2500

## Background

The Design team is releasing a Typography update (sometime in March), which means that the PW team needs to upgrade any of our products where components aren't currently being using, but a V1 or V3 component is available.

If the updates aren't made prior to the Typography update, there's a chance that Veteran-facing issues may occur.

[Income Limits App](https://www.va.gov/health-care/income-limits/introduction)

## Elements that are not using components that should be
1. Link
   - Not all links are using the component, some are though (Intro page does have it, edit links and results screen doesn't)
   - We should be using [the v1 component](https://design.va.gov/components/link/) everywhere we can, including content inside inside accordions

From #17003 

### User story

**AS A** PO/PM for Public Websites products
**I WANT** to implement available V1 or V3 components in features & products currently under Public Websites management where components aren't currently being used
**SO THAT** all PW products are in alignment with the current component version
**AND** Veterans and other users have a consistent experience throughout VA.gov tools and information

## Acceptance criteria

- [x] Links use V1
- [x] other features display correctly
- [x] verify/review the E2E tests
- [x] check desktop and mobile
- [x] a11y review

**Comment by randimays** [comment]
We have some limitations using `<va-link>` in Income Limits:

## Can be converted but not yet

The green and blue action links are not yet converted into web components. Carol from the DST provided a ticket for their upcoming sprint that will add this: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2500

Links that will need to be converted when this is available:

#### Home page
1. <br/><img width="265" alt="Screenshot 2024-02-27 at 10 11 59 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/19175324/2969efc5-b79a-4110-a655-ec4c598852fb">
2. <br/><img width="343" alt="Screenshot 2024-02-27 at 10 12 02 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/19175324/549e81e3-acba-43e1-8b64-04364d8c78a0">

#### Results page

3. <br/><img width="257" alt="Screenshot 2024-02-27 at 10 15 20 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/19175324/2020b0fc-035d-4cb2-99fe-09284d4c6e7a">


## Cannot be converted
Per this Slack conversation: https://dsva.slack.com/archives/C01DBGX4P45/p1692802785182169 the `<va-link>` does not (and will not) support opening links in new tabs. We will have to continue using standard `<a>` tags for these:

#### Results page
1. <br/><img width="492" alt="Screenshot 2024-02-27 at 10 14 54 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/19175324/9216d1c8-f1db-4790-8e84-aa9b7bd4490e">

2. <br/><img width="507" alt="Screenshot 2024-02-27 at 10 16 02 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/19175324/da755edb-8eeb-48d8-80ca-9de66f1a863c">

3. <br/><img width="471" alt="Screenshot 2024-02-27 at 10 16 27 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/19175324/a4e2d5cc-efcf-4e6a-965f-0e31f99ac559">

## Next steps
@jilladams @FranECross Given the above limitations, we can only convert the 3 green/blue action links, and only after the DST ticket is complete. Should we move this ticket to Blocked?

**Comment by randimays** [comment]
@jilladams @FranECross Just flagging here that I edited the comment above after it was posted.

**Comment by FranECross** [comment]
@randimays Thanks for the great update. I agree and am scooting it over to Blocked! cc @jilladams

**Comment by randimays** [comment]
@FranECross To clarify: should I make the updates that I can make now? Or wait until the rest of this is unblocked?

**Comment by FranECross** [comment]
@randimays Which would be easier / better for you? I'm happy to split the ticket so that you can proceed with what you can and then have another ticket for the follow up work that's blocked.

**Comment by randimays** [comment]
@FranECross Let's wait until we are unblocked on the DST ticket and then we can accomplish this all at once, if that's ok.

**Comment by jilladams** [comment]
Removing from S104 since this got blocked, so it doesn't auto get added to S105. Tagging into S106 so we remember to check it. (That's the last sprint on current contract.)

**Comment by FranECross** [comment]
Sounds perfect, thanks!

**Comment by jilladams** [comment]
@FranECross @randimays I reread the comments and I think it's true that we converted the edit links already, and the Action links are outstanding. For reporting's sake, could I ask y'all to collaborate and clarify in ACs what's been done / what's outstanding here?

**Comment by jilladams** [comment]
Ah, nope nevermind. I didn't read far enough. We didn't do any of this yet, fully blocked and waiting on DS team https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2500

**Comment by laflannery** [comment]
In my massive round up of component things I want to note that this is also going to be blocked by a `va-link` [ticket in the DST backlog](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1644) - we need a way to add unique labels on the "Edit" links on the review page. So we will want to keep an eye on that ticket as well before we pull that in

**Comment by FranECross** [comment]
@laflannery Just to clarify your comment above, the unique labels will be 'behind the scenes' for accessibility so that the user knows what they're going to edit? Thanks! Should we have a new ticket to account for that?

**Comment by laflannery** [comment]
This behavior already exists on Prod, I wouldn't expect that we need a new ticket since this would be tested and accounted for when these are updated to use components (i.e. I wouldn't approve this unless I saw that these thinks were behaving the same as they do now). That's why this is blocked until that DST ticket I mentioned is closed.

**Comment by randimays** [comment]
Noting that DST has not yet released this component (though the work has been merged/closed). Once it is available (see releases here: https://github.com/department-of-veterans-affairs/component-library/releases) we can pull this ticket in.

**Comment by randimays** [comment]
Pulling in as stretch for Sprint 6.

**Comment by randimays** [comment]
Successfully validated all changed links in production. Closing as complete.
